daemon/src/lib.rs

```diff
@@ -397,7 +398,6 @@ where
         )
         .create(None)
         .run();
-        std::mem::forget(auto_rollover_address); // leak this address to avoid shutdown
```

Comment:
We are no longer leaking this because we can just keep it around and use it in the tests. Any reason why leaking was chosen over storing it in the actor system, @da-kami? (You may have been part of that work).

Reply:
We did not store it in the actor system before because it was simply not needed (unused). But with using it in the tests it is. that is all good. I think it's fine to store it there for potential future use in production as well.